streamer/src/nonblocking/quic.rs

```diff
@@ -545,12 +568,27 @@ fn run_packet_batch_sender(
     packet_receiver: Receiver<PacketAccumulator>,
     stats: Arc<StreamerStats>,
     cancel: CancellationToken,
+    send_client_id: bool,
 ) {
     let mut channel_disconnected = false;
     trace!("enter packet_batch_sender");
     loop {
         let mut packet_perf_measure: Vec<([u8; 64], Instant)> = Vec::default();
-        let mut packet_batch = BytesPacketBatch::with_capacity(PACKETS_PER_BATCH);
+
+        // Create the appropriate batch type based on configuration
+        let mut regular_batch = if !send_client_id {
+            Some(BytesPacketBatch::with_capacity(PACKETS_PER_BATCH))
+        } else {
+            None
+        };
+        let mut batch_with_client_id = if send_client_id {
+            Some(BytesPacketBatchWithClientId::with_capacity(
+                PACKETS_PER_BATCH,
+            ))
+        } else {
+            None
+        };
```

Comment:
not to self - this is weird to use a bool flag to set TWO different valuables. Feels like "withClientId" is not the same type as BYTES, so maybe it shoudl not be fitted at same level?